jest.config.json

```diff
@@ -8,6 +8,7 @@
     "\\.(jpg|jpeg|png)$": "<rootDir>/__mocks__/file.mock.ts",
     "^@carbon/icons-react/es/(.*)$": "@carbon/icons-react/lib/$1",
     "^carbon-components-react/es/(.*)$": "carbon-components-react/lib/$1",
+    "@openmrs/esm-framework/mock": "@openmrs/esm-framework/mock",
```

Comment:
What is this?

What should happen is that

``` 
"@openmrs/esm-framework": "@openmrs/esm-framework/mock"
```

replaces the old entry.

The old entry should be completely removed. If the `useCurrentPatient` should be overriden or removed this can be done in the setup, or in each relevant test module (would be better in any case, too, as it makes this more transparent).

Reply:
Sure, I thought using global overrides was more ideal since refactoring per test would introduce a lot of "common code" (boilerplate code) across the test suites. 

> (would be better in any case, too, as it makes this more transparent).

Like you said it would be better, let me refactor this :)

Reply:
Yeah if we want to have it global this would work, but then using the test setup would work, too.

I guess at the end its a bit about taste here - but I would love to have it at least in blueprint in an idealized form.

Reply:
I have come to believe this would make more sense to the other side of the world like java:

```java

public abstract class ChartWidgetsMocks extends BaseMocks {
     // Add all your overrides in one place
}

```

I think it's easier to maintain these mocks, if a given test case expects a different behaviour, then it should handle this at the setup level?

Reply:
Hm not sure I understand. The class example confuses me, too.

What is your specific question?

Reply:
Right, @samuelmale , what Florian is suggesting (I think) is that instead of overriding the mock in a local mock file, which is what you've done, you should override the mock in `src/setupTests.ts`. You can do something like

```typescript
import { useCurrentPatient } from "@openmrs/esm-framework/mock"

useCurrentPatient.mockReturnValue([false, mockPatient, mockPatient.id, null]);
```

(assuming that `useCurrentPatient` is already a `jest.Mock`.)

Reply:
Thanks for the insight, updated.